Possible undefined behavior (new warning in release build GCC)

### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

_No response_

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

Arch Linux

### Relevant hardware info

_No response_

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Putting this here so I don't forget to look at this later.  I'm getting a new compile warning with GCC in a release build.

```
[235/415] Compiling C++ object src/dos/libdos.a.p/cdrom_image.cpp.o
In file included from /usr/include/c++/13.2.1/ios:42,
                 from /usr/include/c++/13.2.1/ostream:40,
                 from /usr/include/c++/13.2.1/bits/unique_ptr.h:42,
                 from /usr/include/c++/13.2.1/memory:78,
                 from ../../include/dosbox.h:26,
                 from ../../src/dos/cdrom.h:25,
                 from ../../src/dos/cdrom_image.cpp:24:
In static member function ‘static constexpr std::char_traits<char>::char_type* std::char_traits<char>::copy(char_type*, const char_type*, std::size_t)’,
    inlined from ‘static constexpr void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_S_copy(_CharT*, const _CharT*, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/13.2.1/bits/basic_string.h:420:21,
    inlined from ‘constexpr std::__cxx11::basic_string<_CharT, _Traits, _Allocator>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::_M_replace(size_type, size_type, const _CharT*, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/13.2.1/bits/basic_string.tcc:537:16,
    inlined from ‘constexpr std::__cxx11::basic_string<_CharT, _Traits, _Alloc>& std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::assign(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, size_type, size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’ at /usr/include/c++/13.2.1/bits/basic_string.h:1636:26,
    inlined from ‘bool CDROM_Interface_Image::GetCueString(std::string&, std::istream&)’ at ../../src/dos/cdrom_image.cpp:1449:14:
/usr/include/c++/13.2.1/bits/char_traits.h:445:56: warning: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ accessing 2 or more bytes at offsets 0 and 1 overlaps 1 or more bytes at offset 1 [-Wrestrict]
  445 |         return static_cast<char_type*>(__builtin_memcpy(__s1, __s2, __n));
      |                                        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```

It's complaining about this function (line 1449) and I have complaints about this function too at first glance.  Probably would be good to re-write this function for both readability and to avoid potential UB:

https://github.com/dosbox-staging/dosbox-staging/blob/0c8e8cec90cdbda8821492e1b3094b58d11746cd/src/dos/cdrom_image.cpp#L1443-L1459

### Steps to reproduce the behaviour.

Compile a release build with GCC 13.2.1.

### Download URL of affected game or software

_No response_

### Your configuration

_No response_

### Provide a Log

_No response_

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

--- a/.github/workflows/pvs-studio.yml
+++ b/.github/workflows/pvs-studio.yml
@@ -112,7 +112,7 @@ jobs:
 
       - name: Summarize report
         env:
-          MAX_BUGS: 109
+          MAX_BUGS: 108
         run: |
           echo "Full report is included in build Artifacts"
           echo

--- a/src/dos/cdrom.h
+++ b/src/dos/cdrom.h
@@ -375,7 +375,7 @@ private:
 	} player;
 
 	// Private utility functions
-	bool  LoadIsoFile(char *filename);
+	bool  LoadIsoFile(const char *filename);
 	bool  CanReadPVD(TrackFile *file,
 	                 const uint16_t sectorSize,
 	                 const bool mode2);
@@ -383,7 +383,7 @@ private:
 	void CDAudioCallBack(uint16_t desired_frames);
 
 	// Private functions for cue sheet processing
-	bool  LoadCueSheet(char *cuefile);
+	bool  LoadCueSheet(const char *cuefile);
 	bool  GetRealFileName(std::string& filename, std::string& pathname);
 	bool  GetCueKeyword(std::string &keyword, std::istream &in);
 	bool  GetCueFrame(uint32_t &frames, std::istream &in);

--- a/src/dos/cdrom_image.cpp
+++ b/src/dos/cdrom_image.cpp
@@ -1445,7 +1445,7 @@ bool CDROM_Interface_Image::GetCueString(std::string& str, std::istream& in)
 	in >> str;
 	if (str[0] == '\"') {
 		if (str[str.size() - 1] == '\"') {
-			str.assign(str, 1, str.size() - 2);
+			str = str.substr(1, str.size() - 2);
 		} else {
 			in.seekg(pos, std::ios::beg);
 			char buffer[MAX_FILENAME_LENGTH];

--- a/src/dos/cdrom_image.cpp
+++ b/src/dos/cdrom_image.cpp
@@ -529,7 +529,7 @@ CDROM_Interface_Image::~CDROM_Interface_Image()
 
 bool CDROM_Interface_Image::SetDevice(const char* path, [[maybe_unused]] const int cd_number)
 {
-	const bool result = LoadCueSheet((char *)path) || LoadIsoFile((char *)path);
+	const bool result = LoadCueSheet(path) || LoadIsoFile(path);
 	if (!result) {
 		// print error message on dosbox console
 		char buf[MAX_LINE_LENGTH];
@@ -1062,7 +1062,7 @@ void CDROM_Interface_Image::CDAudioCallBack(uint16_t desired_track_frames)
 	}
 }
 
-bool CDROM_Interface_Image::LoadIsoFile(char* filename)
+bool CDROM_Interface_Image::LoadIsoFile(const char* filename)
 {
 	tracks.clear();
 
@@ -1153,7 +1153,7 @@ static std::string dirname(char* file)
 }
 #endif
 
-bool CDROM_Interface_Image::LoadCueSheet(char *cuefile)
+bool CDROM_Interface_Image::LoadCueSheet(const char *cuefile)
 {
 	tracks.clear();
 

